flight_backend/api/views/flight.py

```diff
@@ -0,0 +1,176 @@
+from rest_framework.response import Response
+from rest_framework import status
+from rest_framework.viewsets import ModelViewSet
+from rest_framework.permissions import IsAuthenticated, IsAdminUser
+from rest_framework.decorators import action
+from django.shortcuts import get_object_or_404
+
+from api.models import Aircraft, Location, Flight, Ticket
+from api.serializers import (
+    LocationSerializer,
+    AircraftSerializer,
+    FlightSerializer,
+    TicketSerializer,
+)
+from api.helpers import validate_flight_details
+from api.constants import STATUSES, AVAILABLE, BOOKED
+from api.tasks import flight_booking_notification, flight_reservation_notification
+
+
+class LocationViewSet(ModelViewSet):
+    serializer_class = LocationSerializer
+    queryset = Location.objects.all()
+    permission_classes = (IsAuthenticated, IsAdminUser)
+
+
+class AircraftViewSet(ModelViewSet):
+    serializer_class = AircraftSerializer
+    queryset = Aircraft.objects.all()
+    permission_classes = (IsAuthenticated, IsAdminUser)
+
+
+class FlightViewSet(ModelViewSet):
+
+    """
+    POST flight/
+    GET flight/
+    GET flight/:pk/
+    PUT flight/:pk/
+    DELETE flight/:pk/
+    """
+
+    serializer_class = FlightSerializer
+    queryset = Flight.objects.all()
+
+    def get_permissions(self):
+        """
+        Set and get permissions for the flight view.
+        """
+
+        if self.action == "status":
+            permission_classes = (IsAuthenticated,)
+        else:
+            permission_classes = (IsAuthenticated, IsAdminUser)
+
+        return [permission() for permission in permission_classes]
+
+    def perform_create(self, serializer):
+        serializer.save()
```

Comment:
is this necessary if there are no overrides?

Reply:
I am doing data validation in the create method and I think its standard procedure to have your serializer.save() in the perform_create method.

Reply:
would [this](https://github.com/encode/django-rest-framework/blob/8935db1be92a2e92860f57cd01ebf11e09a08e9d/rest_framework/mixins.py#L24) not take care of that?

Reply:
Erh, looking at the link, the create and the perform create methods are structured the same way mine is. Maybe I don't really understand what you mean, we can talk more on slack.

Reply:
sorry.. meant the parent class already has `perform_create` defined. not sure if it's needed to be overridden here without any customisation. calling `self.perform_create` should call the one in the parent class.

Reply:
ok, now I understand. Thanks!